rest-dashboard.go

```diff
@@ -165,11 +165,12 @@ func (r *Client) SearchDashboards(query string, starred bool, tags ...string) ([
 // may be only loaded with HTTP API but not created or updated.
 //
 // Reflects POST /api/dashboards/db API call.
-func (r *Client) SetDashboard(board Board, overwrite bool) (StatusMessage, error) {
+func (r *Client) SetDashboard(board Board, folderId int, overwrite bool) (StatusMessage, error) {
```

Comment:
Lets please avoid adding new parameters like that. Could you please add a type `type SetDashboardParams func(values url.Values)` just like in `rest-annotation.go`, convert this into a variadic function, and then iterate over them instead of adding this? We really want to avoid this.